### PR TITLE
Implement user metadata API/CLI

### PR DESCRIPTION
**Pull request by mweibel**

### CODE REVIEW

File: rokka/sourceimages.go [hunk 1]
```diff
@@ -232,3 +232,64 @@ func (c *Client) DeleteDynamicMetadata(org, hash, name string, options DynamicMe
...
+//
+// See: https://rokka.io/documentation/references/image-metadata.html
+func (c *Client) DeleteUserMetadata(org, hash string) error {
+	req, err := c.NewRequest(http.MethodDelete, fmt.Sprintf("/sourceimages/%s/%s/meta/user", org, hash), nil, nil)
+	if err != nil {
+		return err
+	}
+
+	return c.Call(req, nil, nil)
+}
```
Comment (dpacassi) [inline rokka/sourceimages.go:274]: The SetUserMetadata, UpdateUserMetadata and DeleteUserMetadata functions look quite similar.
Maybe we should create a helper function which gets called by these 3 functions with the right parameters in order to reduce redundancy?

File: rokka/sourceimages.go [hunk 2]
```diff
@@ -232,3 +232,64 @@ func (c *Client) DeleteDynamicMetadata(org, hash, name string, options DynamicMe
...
+//
+// See: https://rokka.io/documentation/references/image-metadata.html
+func (c *Client) DeleteUserMetadataByName(org, hash, name string) error {
+	req, err := c.NewRequest(http.MethodDelete, fmt.Sprintf("/sourceimages/%s/%s/meta/user/%s", org, hash, name), nil, nil)
+	if err != nil {
+		return err
+	}
+
+	return c.Call(req, nil, nil)
+}
```
Comment (dpacassi) [inline rokka/sourceimages.go:288]: Same goes for the UpdateUserMetadataByName and DeleteUserMetadataByName functions. Maybe we should create a helper function which gets called by these 2 functions with the right parameters in order to reduce redundancy?

**Review by dpacassi — COMMENTED** [review]
There are two minor redundance issues we might want to tackle before merging. Looks otherwise good to me!

File: rokka/sourceimages.go (hunk 2 above)
Comment (mweibel) [reply rokka/sourceimages.go:288]: fixed, too

File: rokka/sourceimages.go (hunk 1 above)
Comment (mweibel) [reply rokka/sourceimages.go:274]: fixed

**Comment by mweibel** [comment]
@dpacassi I fixed the review comments and added another commit which fixes an issue with the create organization call (plus adds a test for it ;)) --> please don't squash if you merge.

**Pull request merged**